golang/lib/execution/test_suite_executor.go

```diff
@@ -157,41 +154,31 @@ func runTestExecutionFlow(ctx context.Context, testsuite testsuite.TestSuite, co
 		executionClient,
 		filesArtifactUrls)
 
-	// TODO Also time out the setup with the API container rather than storing this locally
-	//  to reduce complexity inside the lib
 	logrus.Info("Setting up the test network...")
 	untypedNetwork, err := test.Setup(networkCtx)
 	if err != nil {
 		return stacktrace.Propagate(err, "An error occurred setting up the test network")
 	}
+	time.Sleep(10 * time.Second)
+	if _, err := executionClient.RegisterTestSetupCompletion(ctx, &emptypb.Empty{}); err != nil {
+		return stacktrace.Propagate(err, "An error occurred registering the test setup completion with the API container")
+	}
 	logrus.Info("Test network set up")
 
 	logrus.Infof("Executing test '%v'...", testName)
+
+	if _, err := executionClient.RegisterTestExecution(ctx, &emptypb.Empty{}); err != nil {
+		return stacktrace.Propagate(err, "An error occurred registering the test execution with the API container")
+	}
 	testResultChan := make(chan error)
 
 	go func() {
 		testResultChan <- runTestInGoroutine(test, untypedNetwork)
```

Comment:
You can get rid of the multithreading entirely - take the contents of `runTestInGoroutine` and inline them here!

Reply:
@mieubrisse if we do that, we miss that nice panic checking that we had in the gofunc (at least, that was my understanding). is that okay?

Reply:
Yep so we definitely need to keep the panic guard - the function wouldn't go away; just be renamed to `runTest` - but you can then replace:
```
go func() {
  testResultChan <- runTestInGoroutine(test, untypedNetwork)
  }
  ```
  with simply `runTest(test, untypedNetwork)` + checking the result of the function